src/Components/Aspire.Milvus.Client/AspireMilvusExtensions.cs

```diff
@@ -0,0 +1,112 @@
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="MilvusClientSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Milvus:Client" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddKeyedMilvusClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MilvusClientSettings>? configureSettings = null)
+    {
+        AddMilvus(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddMilvus(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MilvusClientSettings>? configureSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new MilvusClientSettings();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ParseConnectionString(connectionString);
+        }
+
+        configureSettings?.Invoke(settings);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton(ConfigureMilvus);
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton(serviceKey, (sp, key) => ConfigureMilvus(sp));
+        }
+
+        if (!settings.DisableHealthChecks)
+        {
+            builder.TryAddHealthCheck(new HealthCheckRegistration(
+                serviceKey is null ? "Milvus" : $"Milvus_{connectionName}",
+                sp => new MilvusHealthCheck(serviceKey is null
+                    ? sp.GetRequiredService<MilvusClient>()
+                    : sp.GetRequiredKeyedService<MilvusClient>(serviceKey)),
+                failureStatus: default,
+                tags: default,
+                timeout: default));
+        }
+
+        MilvusClient ConfigureMilvus(IServiceProvider serviceProvider)
+        {
+            if (settings.Endpoint is not null && settings.Key is not null)
+            {
+                return settings.Database is not null
+                    ? new MilvusClient(settings.Endpoint, apiKey: settings.Key!, database: settings.Database, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>())
+                    : new MilvusClient(settings.Endpoint, apiKey: settings.Key!, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>());
```

Comment:
```suggestion
                return new MilvusClient(settings.Endpoint, apiKey: settings.Key, database: settings.Database, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>());
```

From reading the code, I don't think it throws if you pass in `null` for `database`.

Also, you verified `settings.Key is not null` above, you shouldn't need to `!` it.

Reply:
It doesn't need to throw, because it would hit the other conditional and NOT create a client with a database arg.  Good catch on the ! (removed)

Reply:
In Miluvs there is an implied default database, so if you don't specify one (which is NOT a requirement) you write to the default space.

Reply:
What's the functional difference between the code:

```C#
                return settings.Database is not null
                    ? new MilvusClient(settings.Endpoint, apiKey: settings.Key, database: settings.Database, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>())
                    : new MilvusClient(settings.Endpoint, apiKey: settings.Key, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>());
```

and 

```C#
                return new MilvusClient(settings.Endpoint, apiKey: settings.Key, database: settings.Database, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>());
```

?

Decompiling your current code with ILSpy shows:

![image](https://github.com/dotnet/aspire/assets/8291187/8a85fdb0-ab22-49fc-bd50-c991ff8c72f9)

Why check for `null` at all? Just pass `settings.Database` into the ctor.

Reply:
Ah, I missed the ctor that Milvus.Client handled this already...got it, will change/simplify.